lib/functional.js

```diff
@@ -4,4 +4,30 @@ module.exports = (api) => {
 
   api.common.curry = (fn, ...args) => fn.bind(null, ...args);
 
+  api.common.zip = (...arrs) => {
+    if (arrs.length === 0) return [];
+
+    let minLen = arrs[0].length;
+    let i;
+    for (i = 1; i < arrs.length; i++) {
+      minLen = Math.min(arrs[i].length, minLen);
```

Comment:
map/reduce implementation will be so short and beautiful here but need to test performance